dre-service/src/main/java/io/cdap/re/YARERulebookHandler.java

```diff
@@ -80,381 +88,404 @@ public void validateWhen(HttpServiceRequest request, HttpServiceResponder respon
-
-        JsonArray values = new JsonArray();
-        values.add(new JsonPrimitive(id));
-
-        response.add("values", values);
-
-        ServiceUtils.sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
-      } catch (RuleNotFoundException e) {
-        ServiceUtils.error(responder, HttpURLConnection.HTTP_NOT_FOUND, e.getMessage());
-      } catch (Exception e) {
-        ServiceUtils.error(responder, HttpURLConnection.HTTP_INTERNAL_ERROR,
-                           String.format("Unexpected error while updating rule. Please check your request. %s",
-                                         e.getMessage())
-        );
-      }
+        rulesDB.updateRule(namespacedId, rule);
+      });
+
+      JsonObject response = new JsonObject();
+      response.addProperty("status", HttpURLConnection.HTTP_OK);
+      response.addProperty("message", String.format("Successfully updated rule '%s'.", namespacedId.getId()));
+      response.addProperty("count", 1);
+
+      JsonArray values = new JsonArray();
+      values.add(new JsonPrimitive(namespacedId.getId()));
+      response.add("values", values);
+
+      return response;
     });
   }
 
   @GET
-  @Path("rules/{rule-id}")
+  @Path("contexts/{context}/rules/{rule-id}")
   @TransactionPolicy(value = TransactionControl.EXPLICIT)
   public void retrieve(HttpServiceRequest request, HttpServiceResponder responder,
+                       @PathParam("context") String namespace,
                        @PathParam("rule-id") String id, @QueryParam("format") String format) {
-    TransactionRunners.run(getContext(), context -> {
-      try {
-        RulesDB rulesDB = RulesDB.get(context);
-        Map<String, Object> result = rulesDB.retrieveRule(id);
-
-        JsonObject response = new JsonObject();
-        response.addProperty("status", HttpURLConnection.HTTP_OK);
-        response.addProperty("message", String.format("Successfully retrieved rule '%s'.", id));
-        response.addProperty("count", 1);
-
-        if (format == null || format.equalsIgnoreCase("json")) {
-          response.add("values", GSON.toJsonTree(result));
-        } else {
-          JsonArray array = new JsonArray();
-          array.add(new JsonPrimitive(rulesDB.retrieveUsingRuleTemplate(id)));
-          response.add("values", array);
-        }
-
-        ServiceUtils.sendJson(responder, HttpURLConnection.HTTP_OK, response.toString());
-      } catch (RuleNotFoundException e) {
-        ServiceUtils.error(responder, HttpURLConnection.HTTP_NOT_FOUND, e.getMessage());
-      } catch (Exception e) {
-        LOG.debug("Error", e);
-        ServiceUtils.error(responder, HttpURLConnection.HTTP_INTERNAL_ERROR,
-                           String.format("Unexpected error while retrieving rule. Please check your request. %s",
-                                         e.getMessage())
-        );
-      }
-    });
+    respond(request, responder,
+            () -> TransactionRunners.run(getContext(), context -> {
```

Comment:
nit: start this on the previous line, to make it stylistically consistent with the rest of the class.

Reply:
Done